src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java

```diff
@@ -406,6 +418,77 @@ public void deleteOicHearingResult(
         oicHearingResultRepository.delete(oicHearingResult);
     }
 
+    public enum OicSanctionAction {CREATE, UPDATE, QUASH}
+
+    @Transactional
+    @VerifyOffenderAccess
+    public List<Sanction> upsertOicSanctions(
+        final Long adjudicationNumber,
+        final List<OicSanctionRequest> oicSanctionRequests,
+        final OicSanctionAction action) {
+
+        final var adjudication = adjudicationsRepository.findByParties_AdjudicationNumber(adjudicationNumber)
+            .orElseThrow(EntityNotFoundException.withMessage(format("Could not find adjudication number %d", adjudicationNumber)));
+
+        final var hearingResult = oicHearingResultRepository.findByAgencyIncidentIdAndFindingCode(adjudication.getAgencyIncidentId(), FindingCode.PROVED);
+        if (hearingResult.isEmpty()) throw EntityNotFoundException.withMessage(format("Could not find hearing result PROVED for adjudication id %d", adjudication.getAgencyIncidentId()));
+        if (hearingResult.size() > 1) throw EntityNotFoundException.withMessage(format("Multiple PROVED hearing results for adjudication id %d", adjudication.getAgencyIncidentId()));
+
+        Long offenderBookId = adjudication.getOffenderParty().get().getOffenderBooking().getBookingId();
+        Long nextSanctionSeq = oicSanctionRepository.getNextSanctionSeq(offenderBookId);
+
+        List<OicSanction> exitingOicSanctions = oicSanctionRepository.findAllByOicHearingId(hearingResult.get(0).getOicHearingId());
+        switch (action) {
+            case CREATE:
+                if (!exitingOicSanctions.isEmpty()) throw BadRequestException.withMessage(format("Sanctions already exit for adjudication id %d", adjudicationNumber));
+                break;
+            case UPDATE:
+                oicSanctionRepository.deleteAll(exitingOicSanctions);
+                break;
+            case QUASH:
+                if (exitingOicSanctions.isEmpty()) throw EntityNotFoundException.withMessage(format("Cannot quash sanctions, no sanctions found for adjudication id %d", adjudicationNumber));
```

Comment:
Hi Peter, the way the UI works its quite possible we will try and quash an outcome without sanctions, so it would be preferable if this did not throw an error, and just ignore the request, as no sanctions to amend

Reply:
OK, I will remove the exception.